device_parser.c

```diff
@@ -179,6 +180,8 @@ static void parse_board(struct device_parser *dp)
 			dev->fastboot_key_timeout = strtoul(value, NULL, 10);
 		} else if (!strcmp(key, "usb_always_on")) {
 			dev->usb_always_on = !strcmp(value, "true");
+		} else if (!strcmp(key, "ppps_path")) {
+			dev->ppps_path = strdup(value);
```

Comment:
I think you can set `dev->usb` here, thus removing the need for the special case in `device_usb`. I doubt that anybody will mix it with any other usb-supporting control method.

Reply:
I originally did this but I realised it would make the behaviour change depending on the order of the configuration file. So if you specify the ppps_path before qcomlt_dbg, it just wouldn't work for no obvious reason.

Reply:
Yep. I'm too tied to #9. Maybe you are right here.